### PR TITLE
Refactor: `PaymentProcessor` to `TransitProcessor`

### LINKED ISSUE #2235: Model refactor: `PaymentProcessor`
## Acceptance criteria

- [x] Model renamed to `TransitProcessor`
- [x] `TransitAgency.payment_processor` field renamed to `TransitAgency.transit_processor`
- [x] All variable instances renamed to `transit_processor`
- [x] All comments updated to remove mention of "payment"
- [x] `client_id` moved to `TransitAgency.transit_processor_client_id`
- [x] `client_secret_name` moved to `TransitAgency.transit_processor_client_secret_name`
- [x] `audience` moved to `TransitAgency.transit_processor_audience`
- [x] Each field in the model definition has a description
- [x] Existing Django config data is migrated
- [x] Docs are updated

## Impact on the code

- [x] [Code search for `audience` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+audience&type=code)
- [x] [Code search for `client_id` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+client_id&type=code)
- [x] [Code search for `client_secret` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+client_secret&type=code)
- [x] [Code search for `payment_processor` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+payment_processor&type=code)
- [x] [Code search for `PaymentProcessor` in the repo](https://github.com/search?q=repo%3Acal-itp%2Fbenefits+PaymentProcessor&type=code)

## Impact on existing data

- [x] Standard migrations for model/field renames should maintain existing data
- [x] Data migration needed to move field data from old `PaymentProcessor` fields to new `TransitAgency` fields

## Impact on analytics

- [x] The `card_tokenize_url` and `card_tokenize_func` fields are logged in analytics, and are not changed in this scope other than to be on a differently named model
- [ ] `EligibilityType.group_id` is logged in analytics using the [`payment_group` event property](https://github.com/cal-itp/benefits/blob/main/benefits/enrollment/analytics.py#L16). #2249 is a follow-up.

## Impact on Admin permissions

- [x] [View permissions](https://github.com/cal-itp/benefits/blob/main/benefits/core/migrations/0014_staff_group_view_permissions.py) need to be updated for the new model name
- [x] [Edit permissions](https://github.com/cal-itp/benefits/blob/main/benefits/core/migrations/0015_staff_group_edit_permissions.py) need to be updated for the new model name
- [x] [Field exclusions](https://github.com/cal-itp/benefits/blob/main/benefits/core/admin.py#L98) need to be updated to move the newly renamed/moved fields to the `TransitAgencyAdmin`

### PR DESCRIPTION
**Pull request by thekaveman**
Closes #2235 

The PR contains only the basic renames/refactors and no other changes to logic, tests, etc.

`local_fixtures.json` is also updated, you should be able to checkout and run `bin/reset_db.sh` and/or `bin/init.sh` to recreate the local database with these changes.

## Things to check locally

1. Checkout the `main` branch
1. Edit your `local_fixtures.json` (or whatever fixture you are using) to add values to existing `PaymentProcessor` fields updated by this PR (`client_id`, `client_secret_name`, `audience`)
1. Run `bin/reset_db.sh` (now your local DB mirrors the structure of `dev`)
1. Checkout this PR's branch
1. Run `bin/init.sh` (running this PR's migration on top of existing DB structure)

- [ ] For fields that were moved to `TransitAgency`, confirm data is migrated
- [ ] Confirm the `Can X payment processor` permissions are removed from the database
- [ ] Confirm the `Can X transit processor` permissions are added to the database
- [ ] Confirm the Cal-ITP group has `Can [view|change] transit processor` permissions

### CODE REVIEW

**Comment by thekaveman** [comment]
@angela-tran I don't want to step on your toes with #2256 since we both have migrations...

I think this PR is ready to open for review, but I can rebase it onto yours and deal with the migrations conflicts here if you'd rather.

**Comment by angela-tran** [comment]
> @angela-tran I don't want to step on your toes with #2256 since we both have migrations...
> 
> I think this PR is ready to open for review, but I can rebase it onto yours and deal with the migrations conflicts here if you'd rather.

Thanks for checking! I'm still working on my PR, so I'm ok if your PR goes in first and I rebase onto `main` and resolve merge conflicts. I might also look over my migrations when I'm done making changes and see if I can combine them into one migration anyways.

In general, I think it's up to each PR to make sure it's working with the latest migrations in `main`.

File: benefits/core/migrations/0016_refactor_paymentprocessor_transitprocessor.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            name="card_tokenize_func",
+            field=models.TextField(
+                help_text="The function from the card tokenization library to call on the client to initiate the process."
+            ),
+        ),
+        migrations.AlterField(
+            model_name="transitprocessor",
+            name="card_tokenize_url",
+            field=models.TextField(
+                help_text="The abolute URL for the client-side card tokenization library provided by the TransitProcessor."
```
Comment (angela-tran) [inline benefits/core/migrations/0016_refactor_paymentprocessor_transitprocessor.py:121]:
```suggestion
                help_text="The absolute URL for the client-side card tokenization library provided by the TransitProcessor."
```

File: benefits/core/models.py [hunk 2]
```diff
@@ -233,22 +233,19 @@ def by_id(id):
...
-    card_tokenize_func = models.TextField()
-    card_tokenize_env = models.TextField()
-
-    @property
-    def client_secret(self):
-        return get_secret_by_name(self.client_secret_name)
+    name = models.TextField(help_text="Primary internal display name for this TransitProcessor instance, e.g. in the Admin.")
+    api_base_url = models.TextField(help_text="The absolute base URL for the TransitProcessor's API, including https://.")
+    card_tokenize_url = models.TextField(
+        help_text="The abolute URL for the client-side card tokenization library provided by the TransitProcessor."
```
Comment (angela-tran) [inline benefits/core/models.py:243]:
```suggestion
        help_text="The absolute URL for the client-side card tokenization library provided by the TransitProcessor."
```

**Comment by angela-tran** [comment]
### Things to check locally

* [x]  For fields that were moved to `TransitAgency`, confirm data is migrated
* [x]  Confirm the `Can X payment processor` permissions are removed from the database
* [x]  Confirm the `Can X transit processor` permissions are added to the database

<br/>

> * [ ]  Confirm the Cal-ITP group has `Can [add|change] transit processor` permissions

@thekaveman Did you mean `Can [view|change] transit processor`?

* [x]  Confirm the Cal-ITP group has `Can [view|change] transit processor` permissions

**Review by angela-tran — CHANGES_REQUESTED** [review]
Code changes look good 👍 

Noticed a small typo

File: benefits/core/models.py (hunk 2 above)
Comment (thekaveman) [reply benefits/core/models.py:243]: Thanks! [`fb35314`](https://github.com/cal-itp/benefits/pull/2262/commits/fb3531410666cee6ee0aba41749d87d635c3b92e)

File: benefits/core/migrations/0016_refactor_paymentprocessor_transitprocessor.py (hunk 1 above)
Comment (thekaveman) [reply benefits/core/migrations/0016_refactor_paymentprocessor_transitprocessor.py]: Thanks! [8f0f2a7](https://github.com/cal-itp/benefits/pull/2262/commits/8f0f2a7c525e959a843257cbb983cb19ab3f32ff)

**Comment by lalver1** [comment]
Looks good to me too!

- [x] For fields that were moved to `TransitAgency`, confirm data is migrated
- [x] Confirm the `Can X payment processor` permissions are removed from the database
- [x] Confirm the `Can X transit processor` permissions are added to the database
- [x] Confirm the Cal-ITP group has `Can [view|change] transit processor` permissions

**Review by angela-tran — APPROVED** [review]

**Pull request merged**